src/app/test-record/select-test-type/select-test-type.container.ts

```diff
@@ -0,0 +1,46 @@
+import { ChangeDetectionStrategy, Component, OnInit } from '@angular/core';
+import { Store } from '@ngrx/store';
+import { IAppState } from '@app/store/state/app.state';
+import { ActivatedRoute } from '@angular/router';
+import { Observable } from 'rxjs';
+import { TestRecordTestType } from '@app/models/test-record-test-type';
+import { TestTypeCategory } from '@app/models/test-type-category';
+import {
+  getFilteredTestTypeCategories,
+  selectTestTypeById
+} from '@app/store/selectors/VehicleTestResultModel.selectors';
+import { UpdateSelectedTestResultModel } from '@app/store/actions/VehicleTestResultModel.actions';
+import { KeyValue } from '@angular/common';
+import { TreeData } from '@app/models/tree-data';
+
+@Component({
+  selector: 'vtm-select-test-type-container',
+  template: `
+    <ng-container *ngIf="testRecord$ | async as testResultObj">
+      <vtm-select-test-type
+        [testResultObj]="testResultObj"
+        [filteredCategories]="filteredCategories$ | async"
+        (selectTypeHandler)="updateSelectedTestResult($event)"
```

Comment:
use `(testTypeSelected)="testTypeSelectedHandler($event)"`. the action of selecting a test type updates right?